bigtable/bigtable.go

```diff
@@ -52,17 +55,84 @@ const mtlsProdAddr = "bigtable.mtls.googleapis.com:443"
 //
 // A Client is safe to use concurrently, except for its Close method.
 type Client struct {
-	connPool          gtransport.ConnPool
-	client            btpb.BigtableClient
-	project, instance string
-	appProfile        string
+	connPool             gtransport.ConnPool
+	client               btpb.BigtableClient
+	project, instance    string
+	appProfile           string
+	metricsTracerFactory *builtinMetricsTracerFactory
 }
 
 // ClientConfig has configurations for the client.
 type ClientConfig struct {
 	// The id of the app profile to associate with all data operations sent from this client.
 	// If unspecified, the default app profile for the instance will be used.
 	AppProfile string
+
+	// If not set, client side metrics will be collected and exported
+	//
+	// To disable client side metrics, set 'MetricsProvider' to 'NoopMetricsProvider'
+	//
+	// To use a custom meter provider,
+	// 1) use 'WithBuiltIn' to get built-in metrics meter provider options
+	// 2) create meter provider
+	// 3) Set 'MetricsProvider' to the created meter provider
+	MetricsProvider MetricsProvider
+}
+
+// MetricsProvider is a wrapper for built in metrics meter provider
+type MetricsProvider interface {
+	isMetricsProvider()
+}
+
+// NoopMetricsProvider can be used to disable built in metrics
+type NoopMetricsProvider struct{}
+
+func (NoopMetricsProvider) isMetricsProvider() {}
+
+// CustomOpenTelemetryMetricsProvider can be used to collect and export builtin metric with custom meter provider
+type CustomOpenTelemetryMetricsProvider struct {
+	MeterProvider *sdkmetric.MeterProvider
```

Comment:
See other comment below.

Reply:
Removed the public API. Will add it later with modifications